packages/swing-store-lmdb/src/lmdbSwingStore.js

```diff
@@ -164,67 +165,213 @@ function makeSwingStore(dirPath, forceReset = false) {
     delete: del,
   };
 
+  /** @type {Set<number>} */
+  const activeStreamFds = new Set();
+  /** @type {Map<string, number>} */
+  const streamFds = new Map();
+  /** @type {Map<string, string>} */
+  const streamStatus = new Map();
+  let statusCounter = 0;
+
+  const STREAM_START = harden({ offset: 0, itemCount: 0 });
+
+  function insistStreamName(streamName) {
+    assert.typeof(streamName, 'string');
+    assert(
+      streamName.match(/^[-\w]+$/),
+      X`invalid stream name ${q(streamName)}`,
+    );
+  }
+
+  function closefd(fd) {
+    try {
+      fs.closeSync(fd);
+    } catch (e) {
+      // closing an already closed fd is OK, but any other errors are probably bad
+      if (e.code !== 'EBADF') {
+        throw e;
+      }
+    }
+  }
+
+  /**
+   * Close a stream that's open for read or write.
+   *
+   * @param {string} streamName  The stream to close
+   */
+  function closeStream(streamName) {
+    insistStreamName(streamName);
+    const fd = streamFds.get(streamName);
+    if (fd) {
+      closefd(fd);
+      streamFds.delete(streamName);
+      activeStreamFds.delete(fd);
+      streamStatus.set(streamName, 'unused');
+    }
+  }
+
   /**
    * Generator function that returns an iterator over the items in a stream.
    *
    * @param {string} streamName  The stream to read
+   * @param {Object} endPosition  The position of the end of the stream
```

Comment:
can we give this a `typedef` to guard against callers passing in something other than a value they got back out earlier? or.. hm, maybe the fact that this type is shared between the LMDB and Simpler verions makes that hard.

Reply:
The type is defined, but dealing with variant internal details between the two implementations is tricky in TypeScript.  I will toss a question to our flock of TypeScript enthusiasts to see if they have a pattern.

In any case it should still have some kind of insist check to make sure it's well formed.  From the API user's point of view it's an opaque object, but internally it varies between the simple store and the LMDB store so checking seems prudent.